eval.go

```diff
@@ -237,14 +237,28 @@ nextFileMatch:
 		}
 
 		for ; nextDoc < docCount; nextDoc++ {
+			repoID := d.repos[nextDoc]
+			repoMetadata := d.repoMetaData[repoID]
+
 			// Skip tombstoned docs
-			if d.repoMetaData[d.repos[nextDoc]].Tombstone {
+			if repoMetadata.Tombstone {
 				continue
 			}
 
+			if opts.EvaluateFileTombstones {
+
+				if repoMetadata.FileTombstones != nil {
+					fileName := string(d.fileName(nextDoc))
```

Comment:
ouch, this is a lot more work than what we did before. `d.fileName` takes work to lookup if I am not mistaken. and we are allocating a string on the heap. I'm fairly convinced before this is ready to ship for reals, we need to switch to a document ID based representation. However, we can ship this PR before that for testing purposes / minimize the amount of work you are doing.